electrode-ota-server-dao-mariadb/src/dao/MetricDAO.ts

```diff
@@ -86,11 +88,23 @@ export default class MetricDAO extends BaseDAO {
 
     public static async metricsByStatusAfterSpecificTime(connection: IConnection, deploymentKey: string, specificDate: Date): Promise<MetricByStatusOutDTO[]> {
         const depResults = await MetricDAO.query(connection, DeploymentQueries.getDeploymentByKey, [deploymentKey]);
-
         if (!depResults || depResults.length === 0) {
             throw new Error("Not found. no deployment found for key [" + deploymentKey + "]");
         }
-        const metricResults = await MetricDAO.query(connection, MetricQueries.getMetricsByStatusAfterSpecificTime, [depResults[0].id, specificDate]);
+
+        const latestVersions = await PackageDAO.getMostRecentlyPromotedVersions(connection);
```

Comment:
what does all versions have to deal with the deploymentKey?

Reply:
Deployment Key doesn't play any role in metrics

Reply:
Can this add this test:
App 1 versions:   1.0 2.0 3.0 4.0 5.0 6.0
App 2 versions: 1.0 2.0
All versions are PROMOTED.

What happens when I get metric data for App 2?
My understanding of this code is, it'll return versions [4.0, 5.0, 6.0].
Then your going to try to find metrics for App 2, versions [4.0, 5.0, 6.0]

If you understand this issue, please add a test for it.

Reply:
Tweaked this query to retrieve recently promoted versions of the app requested. Also added unit tests as above.

When metrics for App 1 is requested, it retrieves for versions [6.0, 5.0, 4.0] of App-1

When metrics for App 2 is requested, it retrieves for versions [2.0, 1.0] of App-2